cmd/lifecycle/builder.go

```diff
@@ -48,26 +60,160 @@ func (b *buildCmd) Args(nargs int, args []string) error {
 }
 
 func (b *buildCmd) Privileges() error {
-	// builder should never be run with privileges
-	if priv.IsPrivileged() {
+	_, privGroup, _, err := b.readData()
+	if err != nil {
+		return err
+	}
+
+	hasPrivGroups := len(privGroup.Group) > 0
+
+	// builder should never be run with privileges if there aren't any stack buildpacks
+	if !hasPrivGroups && priv.IsPrivileged() {
 		return cmd.FailErr(errors.New("refusing to run as root"), "build")
+	} else if hasPrivGroups && !priv.IsPrivileged() {
+		return cmd.FailErr(errors.New("must run as root"), "build")
```

Comment:
We probably need to explain more in this error message with you must run as root (you passed the privileged flag).